pkg/common/utils/utils.go

```diff
@@ -115,13 +116,22 @@ func GetNamespaceQuotaFromAnnotation(namespaceObj *v1.Namespace) *si.Resource {
 	// retrieve resource quota info from annotations
 	cpuQuota := namespaceObj.Annotations["yunikorn.apache.org/namespace.max.cpu"]
 	memQuota := namespaceObj.Annotations["yunikorn.apache.org/namespace.max.memory"]
-
-	// no quota found
-	if cpuQuota == "" && memQuota == "" {
+	namespaceQuota := namespaceObj.Annotations["yunikorn.apache.org/namespace.quota"]
+
+	// order of annotation preference
+	// 1. namespace.quota
+	// 2. namespace.max.* (Retaining for backwards compatibility. Need to be removed in next major release)
+	if namespaceQuota != "" {
+		var namespaceQuotaMap map[string]string
+		json.Unmarshal([]byte(namespaceQuota), &namespaceQuotaMap)
+		return common.GetResource(namespaceQuotaMap)
+	} else if cpuQuota != "" || memQuota != "" {
+		log.Logger().Warn("Using deprecated namespace.max.* annotations to set cpu and/or memory for namespace.",
```

Comment:
We need to mention the new annotations to use in the message.
We should also log the fact that both exist on the same namespace and we ignore the deprecated one.